lib/matplotlib/backend_bases.py

```diff
@@ -3428,7 +3422,13 @@ def add_tool(self, name, tool, group=None, position=None):
                           'not added')
             return
 
-        self._tools[name] = tool_cls(self, name)
```

Comment:
Fair enough, in that case, I would like to add args and kwargs to ``navigation.add_tool`` to pass on to the tool when it gets instantiated and I will remove the passing of a tool directly to ``ToolbarBase.add_tool``